go.mod

```diff
@@ -222,6 +222,6 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.8.4
+replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.8.5
```

Comment:
Could this replace just be removed since it matches the version in the go.mod?

Reply:
Yes, apparently it can. :-) I've just been updating this line robotically and forgotten why we added it in the first place, but it doesn't seem that it's needed to pin it now: `make modules` doesn't change dependencies if I remove it.

Thanks for noticing this!

Reply:
I remember asking this same question like 2 years ago and found it was necessary but can't remember exactly why. I'll try to track down a discussion but I vaguely remember something failing in e2e without this.

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2989#discussion_r1062892967

Reply:
I vaguely remember now... Thanks for the research @nojnhuh!

Let's see how it fares in CI. I'll restore that line if needed.

Reply:
Yes, thanks for the context! Interestingly enough, it looks like CAPI never did add that or if they did, it's not in the current [go.mod](https://github.com/kubernetes-sigs/cluster-api/blob/main/go.mod).

Reply:
It would make sense CAPI doesn't do this since their main module doesn't import their test module. Though it looks like CAPA recently got rid of this, so maybe we should be ok to remove it too? https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/5061

Reply:
I tried checking out the 1.6 branch of CAPZ and building without the `replace` and things seem to work fine. Maybe a Go change enabled this? It doesn't seem like CAPI has fundamentally changed things since then.